[Lottie Exporter] Added documentation for gradient layers

saurabh-ramola opened the pull request:
Added documentation for Linear Gradient Layer and Radial Gradient Layer.

@AnishGG is this correct?

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,22 @@
++---------------------+---------------------------------------------------------------------------+
+|        Amount       |   layers/shape.json -> "ef" -> effects/gFill.json -> effects/opacity.json |
++---------------------+---------------------------------------------------------------------------+
+|     Blend_method    |Not supported                                                              |
```
I guess this you have supported. Please check again.

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,22 @@
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |    layers/shape.json -> "ef" -> effects/gfill.json                        |
++---------------------+---------------------------------------------------------------------------+
+|       Point 1       |layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"                   |
```
There is no param point1 or point2. Only `p1` and `p2` are there. I see that you have added point1 and point2 params, make a NOTE of these 2 things and why have you added these 2.

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,22 @@
++---------------------+---------------------------------------------------------------------------+
+|        Amount       |   layers/shape.json -> "ef" -> effects/gFill.json -> effects/opacity.json |
++---------------------+---------------------------------------------------------------------------+
+|     Blend_method    |Not supported                                                              |
```
Same as previous blend method. I guess you have supported this.

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,22 @@
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |      layers/shape.json -> "ef" -> effects/gfill.json                      |
++---------------------+---------------------------------------------------------------------------+
+|       Center        |      layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"             |
```
Show how you used center and radius as you have mentioned in the comments in the code: https://github.com/synfig/synfig/pull/1446/files#diff-5a70b0c8e040579359b43c0cfd55b9f9R69

saurabh-ramola replied:
I have added the description for the calculation of gradient start point and gradient end point in the important points section.

saurabh-ramola commented:
> @saurabh-ramola Have a look at the changes requested.

I have incorporated the changes as requested by you. ^__^

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,27 @@
++=====================+===========================================================================+
+|       Z depth       |               Depends on ordering of layers in lottie format              |
++---------------------+---------------------------------------------------------------------------+
+|        Amount       |   layers/shape.json -> "ef" -> effects/gFill.json -> effects/opacity.json |
```
There is not path effects/gFill.json. Please check this.

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,27 @@
++---------------------+---------------------------------------------------------------------------+
+|     Blend_method    |Partially supported                                                        |
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |    layers/shape.json -> "ef" -> effects/gfill.json                        |
```
Same. No effects/gFill.json

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,27 @@
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |    layers/shape.json -> "ef" -> effects/gfill.json                        |
++---------------------+---------------------------------------------------------------------------+
+|       p1            |layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"                   |
```
Here also

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,27 @@
++---------------------+---------------------------------------------------------------------------+
+|       p1            |layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"                   |
++---------------------+---------------------------------------------------------------------------+
+|       p2            |layers/shape.json ->  "ef" -> effects/gFill.json ->  "e"                   |
```
Here also

AnishGG commented on docs/projects/lottie/layers/linear_gradient.rst:
```diff
@@ -0,0 +1,27 @@
+Important points
+----------------
+
+- p1 and p2 parameters are used to calculate the gradient start point -> "s" and the gradient end point -> "e" respectively.
```
Please make a note about `point1` and `point2` also, which you introduced for the rectangle layer.

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,28 @@
++=====================+===========================================================================+
+|       Z depth       |               Depends on ordering of layers in lottie format              |
++---------------------+---------------------------------------------------------------------------+
+|        Amount       |   layers/shape.json -> "ef" -> effects/gFill.json -> effects/opacity.json |
```
No path effects/gFill.json

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,28 @@
++---------------------+---------------------------------------------------------------------------+
+|     Blend_method    |Partially supported                                                        |
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |      layers/shape.json -> "ef" -> effects/gfill.json                      |
```
same

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,28 @@
++---------------------+---------------------------------------------------------------------------+
+|        Gradient     |      layers/shape.json -> "ef" -> effects/gfill.json                      |
++---------------------+---------------------------------------------------------------------------+
+|       Center        |      layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"             |
```
No path effects/gFill.json

AnishGG commented on docs/projects/lottie/layers/radial_gradient.rst:
```diff
@@ -0,0 +1,28 @@
++---------------------+---------------------------------------------------------------------------+
+|       Center        |      layers/shape.json ->  "ef" -> effects/gFill.json ->  "s"             |
++---------------------+---------------------------------------------------------------------------+
+|       Radius        |      layers/shape.json ->  "ef" -> effects/gFill.json ->  "e"             |
```
same

AnishGG commented on docs/projects/lottie/supported.rst:
```diff
@@ -362,6 +362,50 @@ Switch Group Layer
++---------------------+---------------------+---------------------------------------------+
+|       Opacity       |        |tick|       |                  |cross|                    |
++---------------------+---------------------+---------------------------------------------+
+|     Blend_method    |        |cross|      |                  |cross|                    |
```
Please check other supported features and use "Partially supported" and a "dash" for consistency.

AnishGG commented on docs/projects/lottie/supported.rst:
```diff
@@ -362,6 +362,50 @@ Switch Group Layer
++---------------------+---------------------+---------------------------------------------+
+|     Blend_method    |        |cross|      |                  |cross|                    |
++---------------------+---------------------+---------------------------------------------+
+|       Point 1       |        |tick|       |                  |tick|                     |
```
No point1 param

AnishGG commented on docs/projects/lottie/supported.rst:
```diff
@@ -362,6 +362,50 @@ Switch Group Layer
++---------------------+---------------------+---------------------------------------------+
+|       Point 1       |        |tick|       |                  |tick|                     |
++---------------------+---------------------+---------------------------------------------+
+|       Point 2       |        |tick|       |                  |tick|                     |
```
No point2 param

AnishGG commented on docs/projects/lottie/supported.rst:
```diff
@@ -362,6 +362,50 @@ Switch Group Layer
++---------------------+---------------------+---------------------------------------------+
+|       Opacity       |        |tick|       |                  |cross|                    |
++---------------------+---------------------+---------------------------------------------+
+|     Blend_method    ||cross|              |                  |cross|                    |
```
Same as linear gradient's comment

saurabh-ramola commented:
@AnishGG , I have corrected the formatting and added some more information as requested by you. Let me know if I have missed out on anything else ^__^

AnishGG commented:
@saurabh-ramola can you resolve the conflicts.

saurabh-ramola commented:
> @saurabh-ramola can you resolve the conflicts.

Done ^__^

ice0 commented:
Looks like conflict again :(

saurabh-ramola commented:
> Looks like conflict again :(

Corrected it now

AnishGG commented:
Great!
@ice0 seems ready for merging? ^__^

ice0 commented:
Merged. Thank you!
